### ISSUE TITLE
non-boost tutorial-timer.3 has boost references

### ISSUE THREAD

**Issue by Ryder17z** [chriskohlhoff/asio#864]
https://think-async.com/Asio/asio-1.18.2/doc/asio/tutorial/tuttimer3.html

**Comment by iamOgunyinka** [comment]
Yes, but that's not an issue. It used `boost::bind`, which can be easily replaced with `std::bind`.

**Comment by Ryder17z** [comment]
> Yes, but that's not an issue. It used `boost::bind`, which can be easily replaced with `std::bind`.
That just seems to be a lazy excuse to not update the documentation because somebody forgot.

> PS: Non-boost asio does not mean "we won't use Boost at all for any other thing", it simply means "this asio library is stand-alone and not bound to Boost".
That's contradictory and doesn't help the situation. 

Good bye.

**Comment by Ryanf55** [comment]
The tutorials are largely in the C++03 folder. `asio/src/examples/cpp03/tutorial/timer3/timer.cpp`, which was [not added to the STL till C++11](https://en.cppreference.com/w/cpp/utility/functional/bind). 

There are numerous great videos on using `asio` released recently, however, it's not clear why the main tutorials haven't been updated to reflect more recent uses of C++. I am considering updating some of the examples.
